crates/context/config/src/types.rs

```diff
@@ -372,3 +374,30 @@ impl<'a, T: Deserialize<'a>> Signed<T> {
             .map_or(Err(ConfigError::InvalidSignature), |()| Ok(parsed))
     }
 }
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub struct Proposal {
+    pub author_id: Repr<SignerId>,
+    pub actions: Vec<ProposalAction>,
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub enum ProposalAction {
+    ExternalFunctionCall {
+        receiver_id: AccountId,
+        method_name: String,
+        args: Base64VecU8,
+        deposit: NearToken,
+        gas: Gas,
+    },
+    SetNumApprovals {
+        num_approvals: u32,
+    },
+    SetActiveRequestsLimit {
+        active_proposals_limit: u32,
+    },
+    SetContextValue {
+        key: Box<[u8]>,
+        value: Box<[u8]>,
+    },
+}
```

Comment:
would've be better if the context-proxy contract had a base crate that defines primitives